### PR TITLE
New hardware effects for Razer Kraken Kitty Edition

### LINKED ISSUE #1520: Razer Kraken Kitty Edition Missing Chroma options
### Distribution

Gentoo
Openrazer 3.0.1

### Support

- [X] Have you tried rebooting?
- [X] Have you looked at the [Troubleshooting](https://github.com/openrazer/openrazer/wiki/Troubleshooting) page?

Sister issue in Polychromatic here: https://github.com/polychromatic/polychromatic/issues/328#issue-906327255

Hello,

I have a Razer Kraken Kitty Edition Headset. According to the Razer website, and on the Synapse 3 application on Windows, it supports the following Chroma schemes:

- Breath
- Fire
- Spectrum
- Starlight
- Static
- Wave
- Wheel

However, on openrazer (after looking into where the Kitty edition is set up in the accessory driver), there is currently only the following options:

 - None
 - Static
 - Breath

I would like to use Spectrum and have it synced with the rest of my Razer devices. I tried to "force" it using polychromatic-cli by doing the following command: `polychromatic-cli -d headset -o spectrum` however, that gives me the following error message: `Skipping: Razer Kraken Kitty Edition - Setting this option was unsuccessful or is not supported.`

Here are my various version information strings:
OS: Gentoo
Openrazer: 3.0.1 (rebuilt today just in case)
Polychromatic: 0.6.0 (rebuilt today just in case)

Is there any way to enable chroma options that are disabled for some reason on polychromatic despite the device supporting them?

Thanks,
Kusoneko

### PR DESCRIPTION
**Pull request by 999eagle**
As requested in #1520 I've added the starlight, wave and spectrum effects to the Razer Kraken Kitty Edition. Tested those three effects on my headset and all 3 seem to work.

### CODE REVIEW

File: driver/razeraccessory_driver.c [hunk 1]
```diff
@@ -736,6 +738,50 @@ static ssize_t razer_attr_write_mode_breath(struct device *dev, struct device_at
...
+
+        default: // "Random" colour mode
+            report = razer_chroma_extended_matrix_effect_starlight_random(VARSTORE, ZERO_LED, speed);
+            report.transaction_id.id = 0x1F;
+            break;
+        }
+        break;
+
+    default:
+        printk(KERN_WARNING "razeraccessory: Unknown device\n");
```
Comment (z3ntu) [inline driver/razeraccessory_driver.c:774]: Maybe change to this so we can theoretically find the function easier in the future
```
printk(KERN_WARNING "razeraccessory %s: Unknown device\n", __func__);
```

**Review by z3ntu — APPROVED** [review]
Looks sane, thanks! Will merge after the comment is fixed

File: driver/razeraccessory_driver.c (hunk 1 above)
Comment (999eagle) [reply driver/razeraccessory_driver.c:774]: Wouldn't it make more sense to do this for all functions then? As of now, no other function does something like this even though there are quite a few devices in the `razeraccessory` driver which do not support all available effects.

Comment (z3ntu) [reply driver/razeraccessory_driver.c:774]: I guess? The logging in the drivers is really bad anyways so there's a lot that could be improved (not that it makes much of a difference given the general code quality in the driver imo). I'll just merge anyways, it's a minor detail and indeed `__func__` is used nowhere in the repo

**Pull request merged**